tutorials/scripting/gdscript/gdscript_basics.rst

```diff
@@ -2121,6 +2177,136 @@ class resource is done by calling the ``new`` function on the class object::
         var a = MyClass.new()
         a.some_function()
 
+Traits
+------
+
+Traits are collections of behaviors and attributes that classes can use to guarantee
+functionality to themselves and other objects that may be attempting to use them.
```

Comment:
```suggestion
Since Godot 4.x, GDScript supports traits, which are collections of behaviors and attributes
that classes can use to guarantee
functionality to themselves and other objects that may be attempting to use them.
```
If we're doing a "Since Godot 4.x" introduction (which I personally somewhat dislike, but they are common in the docs), here is a better place than the note, IMO.